Cargo.toml

```diff
@@ -138,8 +138,97 @@ xshell = "0.2.5"
 # We need to freeze the version of the crate, as the raw-api feature is considered unstable
 dashmap = { version = "=5.5.3", features = ["raw-api"] }
 
+[workspace.lints.rust]
+rust_2018_idioms = "warn"
+unused_lifetimes = "warn"
+semicolon_in_expressions_from_macros = "warn"
+
 [workspace.lints.clippy]
+# FIXME Remove the tidy test once the lint table is stable
+
+## lint groups
+complexity = { level = "warn", priority = -1 }
+correctness = { level = "deny", priority = -1 }
+perf = { level = "deny", priority = -1 }
+restriction = { level = "allow", priority = -1 }
+style = { level = "warn", priority = -1 }
+suspicious = { level = "warn", priority = -1 }
+
+## allow following lints
+# () makes a fine error in most cases
+result_unit_err = "allow"
+# We don't expose public APIs that matter like this
+len_without_is_empty = "allow"
+# We currently prefer explicit control flow return over `...?;` statements whose result is unused
+question_mark = "allow"
```

Comment:
This one we can probably kick out now, your last PR changed all of them anyways

Reply:
I will remove it in the follow-up PR (currently working on it)

Reply:
https://github.com/rust-lang/rust-analyzer/pull/16404/commits/3cf1358eb81f24d2e2e5394b938280799e4bde5e